src/server/common/helpers/tracing/async-local-storage.js

```diff
@@ -0,0 +1,6 @@
+import { AsyncLocalStorage } from 'node:async_hooks'
+
+const asyncLocalStorage = new AsyncLocalStorage()
```

Comment:
Do we need/want to export this? Feels safer to keep this internal to the tracing

Reply:
This is a simple "cached" singleton as we are using it in a few places.

Its being pulled in for the:
- `tracing` plugin
- logging
- `fetcher`
- `authedFetcher`